glusterd2/commands/snapshot/snapshot-create.go

```diff
@@ -0,0 +1,765 @@
+}
+func undoStoreSnapshotOnCreate(c transaction.TxnCtx) error {
+
+	var snapInfo snapshot.Snapinfo
+	if err := c.Get("snapinfo", &snapInfo); err != nil {
+		return err
+	}
+
+	/*
+		This has to revisit
+	*/
+	volinfo := &snapInfo.SnapVolinfo
+	if err := volgen.DeleteClientSnapVolfile(&snapInfo); err != nil {
+		c.Logger().WithError(err).WithField(
+			"snapshot", snapshot.GetStorePath(&snapInfo)).Warn("failed to delete client volfile of snapshot")
+	}
+	for _, b := range volinfo.GetLocalBricks() {
+		if err := volgen.DeleteBrickVolfile(&b); err != nil {
+			c.Logger().WithError(err).WithField(
+				"brick", b.Path).Warn("failed to delete brick volfile")
+		}
+	}
+
+	if err := snapshot.DeleteSnapshot(&snapInfo); err != nil {
+
+		c.Logger().WithError(err).WithField(
+			"snapshot", snapshot.GetStorePath(&snapInfo),
+		).Warn("Failed to delete snapinfo from store")
+		return err
+	}
+
+	return nil
+}
+
+// storeSnapshot uses to store the volinfo and to generate client volfile
+func storeSnapshot(c transaction.TxnCtx) error {
+
+	var snapInfo snapshot.Snapinfo
+	if err := c.Get("snapinfo", &snapInfo); err != nil {
+		return err
+	}
+	volinfo := &snapInfo.SnapVolinfo
+
+	vol, err := volume.GetVolume(snapInfo.ParentVolume)
+	if err != nil {
+		c.Logger().WithError(err).WithField(
+			"volume", snapInfo.ParentVolume).Debug("storeVolume: failed to fetch Volinfo from store")
+	}
+
+	vol.SnapList = append(vol.SnapList, volinfo.Name)
+	if err := volume.AddOrUpdateVolumeFunc(vol); err != nil {
+		c.Logger().WithError(err).WithField(
+			"volume", vol.Name).Debug("storeVolume: failed to store Volinfo")
+		return err
+	}
+
+	if err := snapshot.AddOrUpdateSnapFunc(&snapInfo); err != nil {
+		c.Logger().WithError(err).WithField(
+			"volume", volinfo.Name).Debug("storeSnapshot: failed to store snapshot info")
+		return err
+	}
+	if err := volgen2.Generate(); err != nil {
+		c.Logger().WithError(err).WithField(
+			"volume", volinfo.Name).Debug("generateVolfiles: failed to generate volfiles")
+		return err
+	}
+
+	return nil
+}
+
+func unmarshalSnapCreateRequest(msg *api.SnapCreateReq, r *http.Request) (int, error) {
+	if err := restutils.UnmarshalRequest(r, msg); err != nil {
+		return 422, gderrors.ErrJSONParsingFailed
+	}
+
+	if msg.VolName == "" {
+		return http.StatusBadRequest, gderrors.ErrEmptyVolName
+	}
+	if msg.SnapName == "" {
+		return http.StatusBadRequest, gderrors.ErrEmptySnapName
+	}
+	return 0, nil
+}
+func updateMntOps(FsType, MntOpts string) string {
+	switch FsType {
+	case "xfs":
+		if len(MntOpts) > 0 {
+			return (MntOpts + ",nouuid")
+		}
+		return "nouuid"
+
+	case "ext4":
+		fallthrough
+	case "ext3":
+		fallthrough
+	case "ext2":
+	default:
+	}
+	return MntOpts
+}
+func populateBrickMountData(volinfo *volume.Volinfo, snapName string) (map[string]snapshot.BrickMountData, error) {
+	nodeData := make(map[string]snapshot.BrickMountData)
+
+	brickCount := 0
+	for _, subvol := range volinfo.Subvols {
+		for _, b := range subvol.Bricks {
```

Comment:
you can do `for _, b := range volinfo.GetLocalBricks()` and skip `uuid.Equal` check